client/lib/desktop/index.js

```diff
@@ -39,42 +39,48 @@ const Desktop = {
 		ipc.on( 'new-post', this.onNewPost.bind( this ) );
 		ipc.on( 'signout', this.onSignout.bind( this ) );
 		ipc.on( 'toggle-notification-bar', this.onToggleNotifications.bind( this ) );
-		ipc.on( 'cookie-auth-complete', this.onCookieAuthComplete.bind( this ) );
 		ipc.on( 'page-help', this.onShowHelp.bind( this ) );
 
-		window.addEventListener( 'message', this.receiveMessage.bind( this ) );
+		this.store = await getReduxStore();
 
 		// Send some events immediatley - this sets the app state
-		this.sendNotificationCount( store.get( 'wpnotes_unseen_count' ) );
+		this.notificationStatus();
 		this.sendUserLoginStatus();
-
-		location( function( context ) {
-			ipc.send( 'render', context );
-		} );
 	},
 
 	selectedSite: null,
 
+	navigate: function( to ) {
+		if ( isNotificationsOpen( reduxGetState() ) ) {
```

Comment:
The store is already in `this.store` and we can call `this.store.getState()`. Only one import from the Redux Bridge module is needed then: `getReduxStore`.

Can it happen that `navigate` is called before the store initialization in the async `init` call is done?

Reply:
@jsnajdr oh, good to know! The earliest `navigate` is called is after `this.sendUserLoginStatus` was executed.